e2e/app/element/base-element.ts

```diff
@@ -1,77 +1,53 @@
 import {ClickOptions, ElementHandle, Page, WaitForSelectorOptions} from 'puppeteer';
+import Container from 'app/container';
 
 /**
  * BaseElement represents a web element in the DOM.
  * It implements useful methods for querying and interacting with this element.
  */
-export default class BaseElement {
+export default class BaseElement extends Container {
 
-  static asBaseElement(page: Page, elem: ElementHandle): BaseElement {
-    return new BaseElement(page, elem);
+  static asBaseElement(page: Page, elementHandle: ElementHandle): BaseElement {
+    const baseElement = new BaseElement(page);
+    baseElement.setElementHandle(elementHandle);
+    return baseElement;
   }
 
-  protected readonly page: Page;
-  protected css: string;
-  protected xpath: string;
   protected element: ElementHandle;
 
-  constructor(aPage: Page, aElement?: ElementHandle) {
-    this.page = aPage;
-    this.element = aElement || undefined;
+  constructor(protected readonly page: Page, xpath?: string) {
```

Comment:
I am curious what the use cases are for this / how you envision its usage. Also - Should the constructor ever be directly called?